File: app/src/main/java/org/nitri/opentopo/overlay/GestureOverlay.java
package org.nitri.opentopo.overlay;

import android.view.MotionEvent;

import org.osmdroid.views.MapView;
import org.osmdroid.views.overlay.Overlay;

public class GestureOverlay extends Overlay {

    GestureCallback gestureCallback;

    public GestureOverlay(GestureCallback gestureCallback) {
        this.gestureCallback = gestureCallback;
    }

    @Override
    public boolean onFling(MotionEvent pEvent1, MotionEvent pEvent2, float pVelocityX, float pVelocityY, MapView pMapView) {
       gestureCallback.onUserMapInteraction();
        return super.onFling(pEvent1, pEvent2, pVelocityX, pVelocityY, pMapView);
    }

    @Override
    public boolean onScroll(MotionEvent pEvent1, MotionEvent pEvent2, float pDistanceX, float pDistanceY, MapView pMapView) {
        gestureCallback.onUserMapInteraction();
        return super.onScroll(pEvent1, pEvent2, pDistanceX, pDistanceY, pMapView);
    }

    public interface GestureCallback {
        void onUserMapInteraction();
    }
}

File: app/src/main/java/org/nitri/opentopo/overlay/GestureOverlay.kt
package org.nitri.opentopo.overlay

import android.view.MotionEvent
import org.osmdroid.views.MapView
import org.osmdroid.views.overlay.Overlay

class GestureOverlay(private var gestureCallback: GestureCallback) : Overlay() {
    override fun onFling(
        pEvent1: MotionEvent,
        pEvent2: MotionEvent,
        pVelocityX: Float,
        pVelocityY: Float,
        pMapView: MapView
    ): Boolean {
        gestureCallback.onUserMapInteraction()
        return super.onFling(pEvent1, pEvent2, pVelocityX, pVelocityY, pMapView)
    }

    override fun onScroll(
        pEvent1: MotionEvent,
        pEvent2: MotionEvent,
        pDistanceX: Float,
        pDistanceY: Float,
        pMapView: MapView
    ): Boolean {
        gestureCallback.onUserMapInteraction()
        return super.onScroll(pEvent1, pEvent2, pDistanceX, pDistanceY, pMapView)
    }

    interface GestureCallback {
        fun onUserMapInteraction()
    }
}
